src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,156 @@
+    private static final String JOBS_PAGINATION_LIST_XPATH = "//a[text()='Last']";
+    private static final String VACANCY_TITLE_XPATH = "//h1[@class='Copy__title']";
+    private static final String VACANCIES_URLS_XPATH = "//a[@data-analytics-label='jobs_listings_title_link']";
+    private static final String VACANCY_DESCRIPTION_BLOCK_XPATH = "//div[@class='ArticleMarkdown']//*";
+    private static final String VACANCY_CITIES_XPATH = "//span[@class='JobsListings__locationDisplayName']";
+    private static final String VACANCY_JOB_TYPE_XPATH = "//p[text()='Job type']";
+    private static final String VACANCY_TEAM_XPATH = "//p[text()='Team']";
+
+    public StripeParser(WebClient htmlWebClient,
+                        CompanyRepository companyRepository,
+                        LocationProcessor locationProcessor,
+                        @Value("${url.stripe}") String url) {
+        super(htmlWebClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected String getCompanyName() {
+        return "Stripe";
+    }
+    @Override
+    public List<Vacancy> getAllVacancies() throws IOException {
+        HtmlPage page = downloadPage(String.format(url, 0));
+        int pageCount = getPageCount(page);
+        List<Vacancy> vacancyList = new ArrayList<>();
+
+        for (int i = 0; i <= pageCount; i++) {
+            page = downloadPage(String.format(url, i));
+            vacancyList.addAll(fillVacancy(
+                    page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+            ));
+        }
+
+        return vacancyList;
+    }
+
+    private int getPageCount(HtmlPage page) {
+        String url = ((DomElement) page.getByXPath(JOBS_PAGINATION_LIST_XPATH).get(0)).getAttribute("href");
+        return Integer.parseInt(url.replaceAll("\\D+", "")) / pageSize;
+    }
+
+    private List<Vacancy> fillVacancy(List<DomNode> urls, List<DomElement> cities) {
+        List<Vacancy> vacancies = new ArrayList<>();
+
+        for (int i = 0; i < urls.size(); i++) {
+            String jobId = urls.get(i).getAttributes().getNamedItem("href").getNodeValue();
+
+            // Skip repeating vacancies and parse locations for them
+            Set<LocationDTO> locationDTOs = new HashSet<>();
+            while (jobId.equals(urls.get(i).getAttributes().getNamedItem("href").getNodeValue())) {
+                locationDTOs.add(createLocationDto(cities.get(i)));
+                i++;
+            }
+
+            // Variable i stopped in incorrect position
+            i--;
+            vacancies.add(createVacancy(createVacancyDTO(jobId, locationDTOs)));
+        }
+
+        return vacancies;
+    }
+
+    @SneakyThrows
+    private VacancyDTO createVacancyDTO(String jobUrl, Set<LocationDTO> locationDTOS) {
+        VacancyDTO vacancyDTO = new VacancyDTO();
```

Comment:
Do you need this DTO at all? What you're doing here is just retrieving some information from a page. Do you really need to put it into DTO first, but then get it from DTO and put it into the entity? Can't you just put it into an entity right away and get rid of DTO?